Split num litter for semgrep parsing

maxmysten opened the pull request:
These changes are necessary to support the semgrep grammar work that I am doing. 

We will need to make more changes to support annotations correctly and scripts.

All tests pass both with this grammar as well as the semgrep additions that will be published next.

maxmysten committed:
Split num litter for semgrep parsing

Now the typed and untyped num lits are distinct.

maxmysten committed:
Replacing dot_expression with _dot_or_index_chain

This mimics more closely how the compiler sees this. Especially with the
recent changes announced, to move; 3+ length do_expressions will occur.

The code prior to this change would treat the dot_expression as the root
of the subtree.  This would lead to issues if the expression did not
have an expression_term in that slot.

This new version puts the call receiving at the root.  This is also
necessary for Semgrep grammer to work correclty.

tzakian commented on grammar.js:
```diff
@@ -29,7 +29,10 @@ const PRECEDENCE = {
   name: 'move',
-  extras: $ => [$._whitespace, $.line_comment, $.block_comment, $.newline, $.annotation],
+  //extras: $ => [$._whitespace, $.line_comment, $.block_comment, $.newline, $.annotation],
+  extras: $ => [$.line_comment, $.block_comment, $.annotation, /\s/],
```
@damirka do you still need empty lines for the formatter?

damirka replied:
I do need!

tzakian commented on grammar.js:
```diff
@@ -41,6 +44,9 @@ module.exports = grammar({
     [$.modifier, $.native_struct_definition],
     [$._expression, $._binary_operand],
   ],
+  inline: $ => [
```
Is this used anywhere? I don't see any usages but maybe I'm missing it.

tzakian commented on grammar.js:
```diff
@@ -760,7 +767,7 @@ module.exports = grammar({
       $.name_expression,
       $.call_expression,
-      $.macro_call_expression,
+      $.macro_call_expression, 
```
Nit: trailing whitespace

tzakian reviewed (commented):
Overall looks good to me except for a couple nits. Main thing is I want to make sure @damirka is ok with these changes/they won't break the autoformatter work before we land it.

damirka commented:
It will break formatter and I'll have to redo some parts. But I def need newline character for the time beimg and it needs to align with my PR

https://github.com/tzakian/tree-sitter-move/pull/8

maxmysten committed:
Readding whitepaces

maxmysten committed:
Readding whitepaces

maxmysten committed:
Merge remote-tracking branch 'origin/main' into maxmysten/tree-sitter-move

maxmysten committed:
Merge branch 'main' of https://github.com/tzakian/tree-sitter-move into tree-sitter-move/maxmysten/tree-sitter-move

maxmysten commented:
> It will break formatter and I'll have to redo some parts. But I def need newline character for the time beimg and it needs to align with my PR
> 
> #8

Cool readded and will remove for semgrep in semgrep grammer.

maxmysten committed:
Fix macro call type arguments.

maxmysten committed:
Fixing field access in grammar

maxmysten committed:
Merge branch 'maxmysten/tree-sitter-move' of https://github.com/maxmysten/tree-sitter-move into maxmysten/tree-sitter-move

maxmysten committed:
Receiver_call don't seem to need type_arguments

maxmysten committed:
Receiver_call will try to match type_arguments wrong

If you have a dot expression followed by a "<", instead of this being
a binary expression, it tires to turn this into an apply_type
or type_arguments and errors out.

maxmysten committed:
Merge branch 'maxmysten/tree-sitter-move' of https://github.com/maxmysten/tree-sitter-move into maxmysten/tree-sitter-move

maxmysten committed:
Added a comment and changes the lang name to move_on_sui.

maxmysten committed:
Merge branch 'tzakian:main' into maxmysten/tree-sitter-move

maxmysten committed:
Fixing parsing grammer so marco params work correctly later in semgrep

maxmysten committed:
Hiding newline from parse tree
